src/utils/ft_split_util.c

```diff
@@ -1,86 +1,73 @@
+#include "../../include/minishell.h"
 
-int	count_size(char **av)
+int	check_white_space(char c)
 {
-	int	i;
-	int	j;
-	int	length;
-
-	i = 1;
-	length = 0;
-	while (0 != av[i])
-	{
-		j = 0;
-		while (0 != av[i][j])
-		{
-			if ((9 <= av[i][j] && 13 >= av[i][j]) || 32 == av[i][j])
-				j++;
-			else
-			{
-				while (0 != av[i][j]
-					&& (9 > av[i][j] || 13 < av[i][j]) && 32 != av[i][j])
-					j++;
-				length++;
-			}
-		}
-		i++;
-	}
-	return (length);
+	if ((c >= 9 && c <= 13) || c == 32)
+		return (1);
+	return (0);
 }
 
-void	make_str(char *word, char *av, int position, int length)
+void	handle_quote(char *line, int *count, int *idx)
 {
-	int		i;
+	int	qoute_num;
 
-	i = 0;
-	while (i < length)
+	qoute_num = line[*idx];
+	(*idx)++;
+	while (line[*idx] != qoute_num)
 	{
-		word[i] = av[position];
-		position++;
-		i++;
+		if (line[*idx] == '\0')
+			exit(0);
+		(*count)++;
+		(*idx)++;
 	}
-	word[i] = '\0';
 }
 
-int	check_size(char **av, int i, int j)
+int	count_split_size(char *str)
 {
-	while ((9 <= av[i][j] && 13 >= av[i][j]) || 32 == av[i][j])
+	int	i;
+	int	length;
+
+	i = 0;
+	length = 0;
+	while (str[i] != '\0')
 	{
-		j++;
-		if (0 == av[i][j])
-			return (0);
+		if (check_white_space(str[i]))
+			i++;
+		else
+		{
+			while (str[i] != 0
+				&& !check_white_space(str[i]))
+				i++;
+			length++;
+		}
 	}
-	while (('0' <= av[i][j] && '9' >= av[i][j])
-	|| '+' == av[i][j] || '-' == av[i][j])
-		j++;
-	return (j);
+	return (length);
 }
 
-int	check_index(char **result, int index)
+int	check_size(char *line, char **str, int *i)
 {
-	if (0 == result[index])
+	int	rtn;
+
+	rtn = 0;
+	while (check_white_space(line[*i]))
 	{
-		index--;
-		return (1);
+		(*i)++;
+		if (line[*i] == 0)
+			return (0);
 	}
-	return (0);
-}
-
-void	create_result(char **result, int *index, int *j, int *position)
-{
-	if (0 != *j)
-		result[*index] = (char *)malloc(sizeof(char) * (*j - *position + 1));
-	else
-		return ;
-	if (0 == result[*index])
+	while (!check_white_space(line[*i]) && line[*i] != '\0')
 	{
-		while (0 <= (*index)--)
+		if (line[*i] == 34 || line[*i] == 39)
 		{
-			free(result[*index]);
-			result[*index] = 0;
+			handle_quote(line, &rtn, i);
+			break ;
 		}
-		free(result);
-		result = 0;
-		ft_putstr_fd("Error\n", 2);
-		exit(1);
+		rtn++;
+		(*i)++;
 	}
+	(*str) = malloc(sizeof(char) * rtn + 1);
+	if (!(*str))
+		exit(1);
```

Comment:
```suggestion
		exit(EXIT_FAILURE);
```

매직넘버를 지양합시다!


FYI
[매직넘버 설명](https://namu.wiki/w/%EB%A7%A4%EC%A7%81%EB%84%98%EB%B2%84#s-4)

Reply:
https://docs.microsoft.com/ko-kr/cpp/c-runtime-library/exit-success-exit-failure?view=msvc-170

홀랭 바꿔서 올리겠습니다

Reply:
위에 보이는 commit 버튼 누르면 github 에서 바로 커밋 가능합니다!

깃허브 꿀팁!

Reply:
@YeonSeong-Lee 해당 함수 exit 만 나중에 custom 화 하면 될 것 같아서 일단 풀리퀘 가능하십니까? 이거 풀리퀘해야 echo 시작 가능